src/cli/project/export/conda.rs

```diff
@@ -0,0 +1,198 @@
+    };
+
+    let channels = environment
+        .channels()
+        .into_iter()
+        .map(|channel| channel.name().to_string())
+        .collect_vec();
+
+    if let VersionSpec::Locked = args.version_spec {
+        let lock_file = project
+            .up_to_date_lock_file(UpdateLockFileOptions {
+                lock_file_usage: args.lock_file_usage.into(),
+                no_install: args.no_install,
+                ..UpdateLockFileOptions::default()
+            })
+            .await?;
+
+        let locked_deps = lock_file
+            .lock_file
+            .environment(environment.name().as_str())
+            .and_then(|env| env.packages(platform).map(Vec::from_iter))
+            .unwrap_or_default();
+
+        let mut dependencies = locked_deps
+            .iter()
+            .filter_map(|d| d.as_conda())
+            .map(|d| CondaEnvDep::Conda(d.package_record().to_string()))
+            .collect_vec();
+
+        let mut pypi_dependencies = locked_deps
+            .iter()
+            .filter_map(|d| d.as_pypi())
+            .filter(|d| !d.is_editable())
+            .map(|d| format!("{}={}", d.data().package.name, d.data().package.version))
+            .collect_vec();
+
+        let editable_dependencies = environment
+            .pypi_dependencies(Some(platform))
+            .into_specs()
+            .filter_map(|(name, spec)| {
+                let requirement = spec
+                    .as_pep508(name.as_normalized(), project.root())
+                    .into_diagnostic()
+                    .unwrap();
+                if let project::manifest::python::RequirementOrEditable::Editable(
+                    _package_name,
+                    requirements_txt,
+                ) = &requirement
+                {
+                    let relative_path = requirements_txt
+                        .path
+                        .as_path()
+                        .strip_prefix(project.manifest_path().parent().unwrap());
+                    return Some(format!("-e ./{}", relative_path.unwrap().to_string_lossy()));
+                }
+                None
+            })
+            .collect_vec();
+
+        pypi_dependencies.extend(editable_dependencies);
+
+        if !pypi_dependencies.is_empty() {
+            dependencies.push(CondaEnvDep::Pip {
+                pip: pypi_dependencies,
+            });
+        }
+
+        let env_file = CondaEnvFile {
+            name: Some(name),
+            channels,
+            dependencies,
+        };
+
+        let env_string = serde_yaml::to_string(&env_file).into_diagnostic()?;
+        println!("{}", env_string);
+
+        return Ok(());
+    }
+
+    let mut dependencies = environment
+        .dependencies(None, Some(platform))
+        .into_specs()
+        .map(|(name, spec)| match args.version_spec {
+            VersionSpec::Manifest => {
+                CondaEnvDep::Conda(MatchSpec::from_nameless(spec, Some(name)).to_string())
```

Comment:
Why would a user not want to add the given match spec?

Reply:
I'm thinking about the range of ways that folks currently use `environment.yml`. I see a lot of `environment.yml`s passed around without any versions specified, and smaller numbers with just the direct dependencies constrained, and lesser again that have the fully resolved dependencies.

I'm trying to provide an option for those non version folks, by giving them an option (as well as to loosen constraints for various testing scenarios), while nudging them towards at least including the direct dependency match specs as default.

At the same time being able to support export the exact dependencies resolved, which is currently a separate code path, but I'd like to clean that up.

Reply:
I see what you mean, however this might not be the right place to introduce this loosing if dependencies logic. For instance there is another Issue talking about this here: https://github.com/prefix-dev/pixi/issues/639. 

If this would be a pixi wide solution which export would benefit from. I'd personally like to start with a `match-spec` or `locked` environment export mechanism. And later we could possibly add the `--pin` style here as well.  

What do you think of that?